apis/kueue/v1alpha1/workload_webhook.go

```diff
@@ -105,6 +105,25 @@ func ValidateWorkload(obj *Workload) field.ErrorList {
 }
 
 func ValidateWorkloadUpdate(newObj, oldObj *Workload) field.ErrorList {
-	return append(ValidateWorkload(newObj),
-		apivalidation.ValidateImmutableField(newObj.Spec.PodSets, oldObj.Spec.PodSets, field.NewPath("spec", "podSets"))...)
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateWorkload(newObj)...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PodSets, oldObj.Spec.PodSets, field.NewPath("spec", "podSets"))...)
+	allErrs = append(allErrs, validateQueueNameUpdate(newObj.Spec.QueueName, oldObj.Spec.QueueName)...)
+	allErrs = append(allErrs, validateAdmissionUpdate(newObj.Spec.Admission, oldObj.Spec.Admission)...)
+
+	return allErrs
+}
+
+func validateQueueNameUpdate(new, old string) field.ErrorList {
+	if len(old) == 0 {
+		return field.ErrorList{}
+	}
+	return apivalidation.ValidateImmutableField(new, old, field.NewPath("spec", "queueName"))
+}
+
+func validateAdmissionUpdate(new, old *Admission) field.ErrorList {
```

Comment:
ditto

// validate that admission is set once

Reply:
Done :)

Reply:
thanks, but the wording is not correct :); can you just use the wording as suggested?